### DISCUSSION TITLE
Extend CI pipeline to check code quality

### DISCUSSION THREAD

**Discussion by grubersjoe** [usebruno/bruno#675] [General]
Hey. Since a lot of different people are contributing now (yay 🎉) I think it might make sense to think about extending the CI pipeline (GitHub actions) to check certain code quality things before merging it.

I for example have encountered several times that the `package-lock.json` is not up-to-date. People simply forget to add it to their commit. Another check could be if the code is correctly formatted and matches the Prettier configuration. Adding a linter like ESLint would be another good way to statically find issues in the code (e.g. unused variables etc.).

In general I think more pipeline steps would be really helpful to ensure certain code quality/style guidelines are met by all contributors. If the pipeline fails, people immediately see what needs correction. Only after the pipeline is green, a PR should be able to be merged in my opinion.

What do you think? We could start with a few simple checks like if the `package-lock.json` is up-to-date and the Prettier code style and add more things if desired later.

**Comment by rabestro** [comment]
Also, since this is an Open Source project, we can add additional services for Quality Gate.
